app/javascript/toolbar-actions/util.spec.js

```diff
@@ -0,0 +1,43 @@
+import { convertMultParamsToRailsMultParams } from './util';
+
+const TESTCASES = [
+  [ // Example Case
+    "selectedUser=all&approvalStateCheckboxes=pending_approval&approvalStateCheckboxes=approved&approvalStateCheckboxes=denied&types=all&selectedPeriod=7&reasonText=",
+    "selectedUser=all&approvalStateCheckboxes[]=pending_approval&approvalStateCheckboxes[]=approved&approvalStateCheckboxes[]=denied&types=all&selectedPeriod=7&reasonText="
+  ], [
+    "foo=bar",
+    "foo=bar"
+  ], [
+    "foobar",
+    "foobar=undefined"
```

Comment:
Is undefined the actual string undefined or like a javascript undefined. If the former, let's pick some other string, because it's confusing as is.

Reply:
If the latter, I don't think undefined values should bleed through.

Reply:
It's the actual string undefined. See:

![Screen Shot 2021-09-21 at 4 05 44 PM](https://user-images.githubusercontent.com/29209973/134240006-93e3940b-dcc1-4c67-8af9-2a6175308a8a.png)

If you give a test of simply `foobar` the code will break it down into `foobar=undefined`.

Reply:
Update: In adding the smarter handling, the code no longer throws in the `undefined` string.